### ISSUE TITLE
Exception in new version when `DOTNET_ENVIRONMENT` environment variable is set to "Development"

### ISSUE THREAD

**Issue by dracan** [BrighterCommand/Brighter#2514]
### Describe the bug

Exception is thrown in new version when `DOTNET_ENVIRONMENT` environment variable is set to "Development".

### To Reproduce

* Create a new dotnet project... `dotnet new worker`
* Do a bare-minimal Brighter "hello world" example...

```csharp
var host = Host.CreateDefaultBuilder()
    .ConfigureServices(services =>
    {
        services.AddBrighter().AutoFromAssemblies();
    })
    .UseConsoleLifetime()
    .Build();

var commandProcessor = host.Services.GetService<IAmACommandProcessor>();
commandProcessor.Send(new MyCommand("Bob"));

host.WaitForShutdown();
```

```csharp
internal class MyCommand : Command
{
    public MyCommand(string name) : base(Guid.NewGuid())
    {
        Name = name;
    }

    public string Name { get; }
}
```

```csharp
internal class MyHandler : RequestHandler<MyCommand>
{
    [RequestLogging(step: 1, timing: HandlerTiming.Before)]
    public override MyCommand Handle(MyCommand command)
    {
        Console.WriteLine($"Hello {command.Name}");
        return base.Handle(command);
    }
}
```

* Run it, and notice the following exception...

```
System.AggregateException: Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer Lifetime: Singleton ImplementationType: Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer': Unable to resolve service for type 'Paramore.Brighter.Transforms.Storage.IAmAStorageProviderAsync' while attempting to activate 'Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer'.)
 ---> System.InvalidOperationException: Error while validating the service descriptor 'ServiceType: Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer Lifetime: Singleton ImplementationType: Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer': Unable to resolve service for type 'Paramore.Brighter.Transforms.Storage.IAmAStorageProviderAsync' while attempting to activate 'Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer'.
 ---> System.InvalidOperationException: Unable to resolve service for type 'Paramore.Brighter.Transforms.Storage.IAmAStorageProviderAsync' while attempting to activate 'Paramore.Brighter.Transforms.Transformers.ClaimCheckTransformer'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(ServiceDescriptor serviceDescriptor, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ValidateService(ServiceDescriptor descriptor)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ValidateService(ServiceDescriptor descriptor)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider..ctor(ICollection`1 serviceDescriptors, ServiceProviderOptions options)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.DependencyInjection.ServiceProvider..ctor(ICollection`1 serviceDescriptors, ServiceProviderOptions options)
   at Microsoft.Extensions.DependencyInjection.ServiceCollectionContainerBuilderExtensions.BuildServiceProvider(IServiceCollection services, ServiceProviderOptions options)
   at Microsoft.Extensions.DependencyInjection.DefaultServiceProviderFactory.CreateServiceProvider(IServiceCollection containerBuilder)
   at Microsoft.Extensions.Hosting.Internal.ServiceFactoryAdapter`1.CreateServiceProvider(Object containerBuilder)
   at Microsoft.Extensions.Hosting.HostBuilder.InitializeServiceProvider()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at Program.<Main>$(String[] args) in C:\Code\BrighterTest2\Program.cs:line 4
```

* Open up `launchsettings.json`, and remove the `DOTNET_ENVIRONMENT` environment variable entry.
* Run again, and it now works as expected.

### Further technical details
- Brighter version: 9.4.0
- Include the output of `dotnet --info`
- The OS (Window, Linux, MacOs, etc)

```
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.200 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.309 [C:\Program Files\dotnet\sdk]
  6.0.400-preview.22301.10 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by iancooper** [comment]
@dracan Thanks for the bug report and repo. As discussed I think its failing because when the environment is Development, .NET is checking the ioc dependency tree and finds that we have registered a ClaimCheckTransformer, but your code has not registered a storage provider. This is quite reasonable on your part because you are not using a ClaimCheck.

I think we need to decide if we need to deiced between only registering the transfomer if you register a storage provider or register a NullStorageProvider that you override with your own storage provider if you need one to avoid this problem

**Comment by iancooper** [comment]
As a workaround you can disable the check. See https://stackoverflow.com/questions/49149065/how-do-i-validate-the-di-container-in-asp-net-core

**Comment by iancooper** [comment]
@dracan It is possible that the behaviour here changed in .net7 as even adding a launch file with the ASP.NET Development flag won't make this happen on our built in examples, which are .net6.

It needs fixing either way, but I suspect this is why it has not popped out on us.

**Comment by iancooper** [comment]
Resolved by #2516

**Comment by iancooper** [comment]
@dracan Should be out in 9.4.1
